### PR TITLE
test: fork child node processes

### PR DESCRIPTION
**Pull request by heisian**
it blends

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ node_js:
   - '8'
   - '6'
 before_install:
-  - if [ "$TRAVIS_BRANCH" == "master" ]; then echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" > .npmrc; fi
+  - if [ "$TRAVIS_BRANCH" == "master" ]; then export _authToken=$NPM_TOKEN; fi
```
Comment (heisian) [inline .travis.yml]: @remy had to do this to get it to build

**Comment by heisian** [comment]
booyakashah!

**Comment by remy** [comment]
👍

**Comment by lzl124631x** [comment]
Cool. Looking forward to merging it!

**Comment by remy** [comment]
I'm just waiting on another PR to be finished and I'll merge this in.

**Comment by remy** [comment]
Can you merge [this commit](https://github.com/remy/nodemon/commit/8967c02301cbc4f1f0cecab2e192e0ede82463cf) into your branch? It should fix the npm install problem properly now and will allow me to merge without having to edit your code.

**Comment by remy** [comment]
Damn, just hit the first issue that says using `fork` is a problem.

Try running `nodemon --inspect app.js` (app.js can be just a `console.log`). Because node isn't spawned, it doesn't open the debugger. Not sure if this is possible with `fork`?

File: lib/monitor/run.js [hunk 2]
```diff
@@ -82,7 +83,17 @@ function run(options) {
     });
   }
 
-  child = spawn.apply(null, spawnArgs);
+  if (executable === 'node') {
+    var forkArgs = cmd.args.slice(1);
```
Comment (remy) [inline lib/monitor/run.js]: Out of interest - why is this `.slice(1)`?

File: lib/monitor/run.js [hunk 3]
```diff
@@ -82,7 +83,17 @@ function run(options) {
     });
   }
 
-  child = spawn.apply(null, spawnArgs);
+  if (executable === 'node') {
```
Comment (remy) [inline lib/monitor/run.js]: I'm also adding `cmd.args[0].indexOf('-') === -1 &&` to this line so that spawns if there's a node argument being passed.

**Pull request merged**